clang/test/SemaSYCL/reqd-work-group-size-device.cpp

```diff
@@ -118,6 +115,10 @@ int main() {
     });
 
 #endif
+    // Ignore duplicate attribute.
+    h.single_task<class test_kernel11>(
+        []() [[sycl::reqd_work_group_size(2, 2, 2),
+               sycl::reqd_work_group_size(2, 2, 2)]] {});
```

Comment:
Do we actually need all the tests in this file now? Other than the default values tests, is there anything in this file which is not already tested in on of the other reqd-work-group test files i.e. - 

```
clang/test/SemaSYCL/intel-reqd-work-group-size-device-direct-prop.cpp
clang/test/SemaSYCL/intel-reqd-work-group-size-device.cpp
clang/test/SemaSYCL/redeclaration-attribute-propagation.cpp
clang/test/SemaSYCL/reqd-work-group-size-device-direct-prop.cpp
clang/test/SemaSYCL/reqd_work_group_size.cpp
```

I feel like we're repeating a lot of tests across files, making it difficult to follow what is actually being tested and if we have sufficient coverage.

Reply:
I will prefer not to make any changes to the existing tests on this PR. The patch is already big . I agree with you, @elizabethandrews that we have many tests regarding the same attribute. Some of them are due to different behaviors on SYCL 2017 vs SYCL 2020. The new test i added about the missing cases. I will work on a separate PR to split AST and diagnostics parts. This will remove some of those redundant tests. Would it be OK? 

Existing issue: https://github.com/intel/llvm/issues/4118

Reply:
Sounds good to me!